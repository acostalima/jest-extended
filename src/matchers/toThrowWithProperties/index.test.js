import matcher from './';

expect.extend(matcher);

const fixtureProperties = { foo: 'foo', bar: 'bar' };

describe('.toThrowWithProperties', () => {
  test('passes when error contains properties', () => {
    const error = Object.assign(new Error('error'), fixtureProperties);

    expect(() => {
      throw error;
    }).toThrowWithProperties(fixtureProperties);
  });

  test('fails when error does not contain properties', () => {
    expect(() =>
      expect(() => {
        throw new Error('error');
      }).toThrowWithProperties(fixtureProperties)
    ).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toThrowWithProperties', () => {
  test('passes when error does not contain properties', () => {
    const error = Object.assign(new Error('error'), { foo: 'foo' });

    expect(() => {
      throw error;
    }).not.toThrowWithProperties(fixtureProperties);
  });

  test('fails when error contains properties', () => {
    const error = Object.assign(new Error('error'), fixtureProperties);

    expect(() =>
      expect(() => {
        throw error;
      }).not.toThrowWithProperties(fixtureProperties)
    ).toThrowErrorMatchingSnapshot();
  });
});
