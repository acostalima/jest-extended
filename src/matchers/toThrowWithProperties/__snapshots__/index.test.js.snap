// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toThrowWithProperties fails when error contains properties 1`] = `
"<dim>expect(</><red>function</><dim>).not.toThrowWithProperties(</><green>properties</><dim>)</>

Expected not to throw with:
  <green>{\\"bar\\": \\"bar\\", \\"foo\\": \\"foo\\"}</>
Thrown with:
  <red>{\\"bar\\": \\"bar\\", \\"foo\\": \\"foo\\"}</>
"
`;

exports[`.toThrowWithProperties fails when error does not contain properties 1`] = `
"<dim>expect(</><red>function</><dim>).toThrowWithProperties(</><green>properties</><dim>)</>

Expected to throw with:
  <green>{\\"bar\\": \\"bar\\", \\"foo\\": \\"foo\\"}</>
Thrown with:
  <red>{}</>
"
`;
